app/src/main/java/com/islaharper/dawnofandroid/presentation/components/MessageBar.kt

```diff
@@ -0,0 +1,166 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Warning
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.R
+import com.islaharper.dawnofandroid.domain.model.MessageBarState
+import kotlinx.coroutines.delay
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+
+@Composable
+fun MessageBar(messageBarState: MessageBarState) {
+    var startAnimation by remember { mutableStateOf(false) }
+    var errorMessage by remember { mutableStateOf("") }
+
+    val socketErrorMessage = stringResource(id = R.string.socket_error_message)
+    val connectErrorMessage = stringResource(id = R.string.connect_error_message)
+
+    LaunchedEffect(key1 = messageBarState) {
+        if (messageBarState.error != null) {
+            errorMessage = when (messageBarState.error) {
+                is SocketTimeoutException -> {
+                    socketErrorMessage
+                }
+
+                is ConnectException -> {
+                    connectErrorMessage
+                }
+
+                else -> {
+                    "${messageBarState.error.message} "
+                }
+            }
+        }
+        startAnimation = true
+        delay(3000)
+        startAnimation = false
+    }
+
+    AnimatedVisibility(
+        visible = messageBarState.error != null && startAnimation ||
+            messageBarState.message != null && startAnimation,
+        enter = expandVertically(
+            animationSpec = tween(300),
+            expandFrom = Alignment.Top,
+        ),
+        exit = shrinkVertically(
+            animationSpec = tween(300),
+            shrinkTowards = Alignment.Top,
+        ),
+    ) {
+        Message(
+            messageBarState = messageBarState,
+            errorMessage = errorMessage,
+        )
+    }
+}
+
+@Composable
+fun Message(
+    messageBarState: MessageBarState,
+    errorMessage: String = "",
+) {
+    val messageBarColour = if (messageBarState.error != null) {
```

Comment:
This composable has a lot of switching depending on whether it has an error state or not. Is there any possibility of decomposing the problem in a different way to create simpler Composables, even if it ends up with a bit of duplication?

Alternatively, you can make some of the logic here (like for instance the message bar content color, and the icon) into parameters for the Composable. It doesn't matter if the Composable has a lot of params, especially if you add sensible defaults. Doing it this way means the complexity is pulled down into the ViewModel

Reply:
Committed a change with some default parameters. I think I'm on the same page as what you were suggesting...

Reply:
Thanks, the commit is definitely an improvement 🎉 
Can we go through this together in our meeting on Monday? I will extend the meeting to an hour